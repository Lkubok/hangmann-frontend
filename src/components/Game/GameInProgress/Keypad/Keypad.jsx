import React, { Component } from "react";
import Keyboard from "react-simple-keyboard";
import { connect } from "react-redux";
import { pressLetter } from "../../../../actions/gameActions";
import "react-simple-keyboard/build/css/index.css";
import * as selectors from "../../../../reducers/selectors";

class Keypad extends Component {
  onKeyPress = button => {
    const { pressLetter, gameId } = this.props;
    pressLetter(button, gameId);
  };

  render() {
    return (
      <Keyboard
        onKeyPress={button => this.onKeyPress(button)}
        layout={{
          default: ["q w e r t y u i o p", "a s d f g h j k l", "z x c v b n m"]
        }}
      />
    );
  }
}
const mapStateToProps = state => ({
  gameId: selectors.getGameId(state)
});
const mapDispatchToProps = {
  pressLetter
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Keypad);
