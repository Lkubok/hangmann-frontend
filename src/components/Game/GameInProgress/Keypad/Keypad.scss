@import "../../../../scss/mixins.scss";
@import "../../../../scss/variables.scss";

.key-button {
  &-typed {
    background-color: red !important;
  }
  &-untyped {
    background-color: green !important;
  }
}
.key-buton-untyped {
  background-color: blue;
}
