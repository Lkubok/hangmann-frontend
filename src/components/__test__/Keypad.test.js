import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Keypad } from "../Game/GameInProgress/Keypad/Keypad";
Enzyme.configure({ adapter: new Adapter() });

describe("KeyPad component", () => {
  let props = {
    guessed: ["a", "b"],
    typed: ["a", "b"],
    gameLang: "pl"
  };
  beforeEach(() => {});
  it("Should match snapshot", () => {
    const wrapper = shallow(<Keypad {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
