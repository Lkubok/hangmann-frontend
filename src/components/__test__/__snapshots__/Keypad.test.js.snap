// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyPad component Should match snapshot 1`] = `
<div
  className="keyboard"
  key="a-b"
>
  <e
    buttonTheme={
      Array [
        Object {
          "buttons": "",
          "class": "key-button key-button-typed-bad",
        },
        Object {
          "buttons": "a b",
          "class": "key-button key-button-typed-good",
        },
        Object {
          "buttons": "q w e r t y u i o p s d f g h j k l z x c v n m ą ę ź ó ł ń ć ś ż",
          "class": "key-button key-button-untyped",
        },
      ]
    }
    layout={
      Object {
        "default": Array [
          "q w e r t y u i o p",
          "a s d f g h j k l",
          "z x c v b n m",
          "ą ę ź ó ł ń ć ś ż",
        ],
      }
    }
    onKeyPress={[Function]}
  />
</div>
`;
